tools/utils.py

```diff
@@ -21,13 +21,23 @@ def wait_for_droplet_creation(droplet):
         print(f'   Exception: {err}')
         raise
 
+def wait_for_droplet_ip(droplet, timeout_seconds=None):
+    start_time = datetime.datetime.now()
+    while timeout_seconds is None \
+            or check_timeout(start_time, timeout_seconds) is not STATUS_TIMEOUT:
+        if droplet.load().ip_address is not None:
+            return STATUS_OK
+        time.sleep(2)
+    return STATUS_TIMEOUT
```

Comment:
maybe we can add a log here to see if the timeout is respected. `502 Bad Gateway` are sometimes caused by too many requests to handle on a server

Reply:
It should be printed here:
```
    else:
        print('   Timeout waiting for IP check')
        destroy_droplet_and_exit(droplet)
```
This loop does not make any request but I will check where the request is made for the `502 Bad Gateway`

Reply:
To `AWS` Api not to the server itself I think

Reply:
I meant just a request haha, It might be the AWS server where their API is running that returned a bad getaway upon receiving to many requests